yarn-project/noir-contracts/src/contracts/child_contract/src/main.nr

```diff
@@ -24,4 +25,10 @@ contract Child {
 
         context.finish(inputs)
     }
+
+    /// ABI pubValue type "open"
+    /// ABI pubValue params [{"name":"args","type":{"kind":"array","length":8,"type":{"kind":"field"}},"visibility":"public"}]
+    fn pubValue(_call_context: pub CallContext, args: pub [Field; dep::aztec3::abi::MAX_ARGS]) -> pub Field {
+        args[0] + 42
+    }
```

Comment:
Check out how this is done now in private functions! Through the ABI definition we can keep meaningful arguments even in child contracts, by asking about the size in fields of the arguments so that way we know how many fields we need to write in the initial witness

Reply:
If you want we can pair on it and I explain to you how we do it :D

Reply:
Discussed via DM. In public functions we don't have the ABI, only the bytecode, so we cannot do this.